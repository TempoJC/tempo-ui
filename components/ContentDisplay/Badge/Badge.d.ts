import { BadgeProps } from './types';
export declare const BADGE_COLOR_COCONUT = "coconut";
export declare const BADGE_COLOR_BLACKBERRY = "blackberry";
export declare const BADGE_COLOR_CHERRY = "cherry";
export declare const BADGE_COLOR_APRICOT = "apricot";
export declare const BADGE_COLOR_KIWI = "kiwi";
export declare const BADGE_COLOR_MINT = "mint";
export declare const BADGE_COLOR_WATER = "water";
export declare const BADGE_COLOR_BLUEBERRY = "blueberry";
export declare const BADGE_COLOR_GRAPE = "grape";
export declare const BADGE_COLOR_WATERMELON = "watermelon";
/**
 * Component to render a badge (with optional text content).
 */
declare const Badge: ({ className, color, kind, size, variant, testId, text, }: BadgeProps) => import("react/jsx-runtime").JSX.Element;
export default Badge;
