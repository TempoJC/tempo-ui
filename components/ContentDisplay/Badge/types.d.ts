export declare const BADGE_COLOR_COCONUT = "coconut";
export declare const BADGE_COLOR_BLACKBERRY = "blackberry";
export declare const BADGE_COLOR_CHERRY = "cherry";
export declare const BADGE_COLOR_APRICOT = "apricot";
export declare const BADGE_COLOR_KIWI = "kiwi";
export declare const BADGE_COLOR_MINT = "mint";
export declare const BADGE_COLOR_WATER = "water";
export declare const BADGE_COLOR_BLUEBERRY = "blueberry";
export declare const BADGE_COLOR_GRAPE = "grape";
export declare const BADGE_COLOR_WATERMELON = "watermelon";
export interface BadgeProps {
    /**
     * Additional CSS to be applied to the component.
     */
    className?: string;
    /**
     * Sets the color scheme of the component.
     */
    color?: "light" | "dark";
    /**
     * Kind of badge. Circular badges only for one character content.
     */
    kind?: "circular" | "autolayout";
    /**
     * Size.
     */
    size?: "large" | "small";
    /**
     * Text.
     */
    text?: string;
    /**
     * Color variant.
     */
    variant?: typeof BADGE_COLOR_COCONUT | typeof BADGE_COLOR_BLACKBERRY | typeof BADGE_COLOR_CHERRY | typeof BADGE_COLOR_APRICOT | typeof BADGE_COLOR_KIWI | typeof BADGE_COLOR_MINT | typeof BADGE_COLOR_WATER | typeof BADGE_COLOR_BLUEBERRY | typeof BADGE_COLOR_GRAPE | typeof BADGE_COLOR_WATERMELON;
    /**
     * Adds a `data-testid` attribute to the component outermost wrapper, set to the given value.
     */
    testId?: string;
}
