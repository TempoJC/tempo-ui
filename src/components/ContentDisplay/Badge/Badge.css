.tempo-badge {
  width: var(--size4);
  height: var(--size4);
  padding: 0;
}

.tempo-badge--small {
  display: flex;
  justify-content: center;
  align-items: center;
}

.tempo-badge__item {
  background: var(--color-token-light-background-inverse);
  font: var(--typography-detail-m-light-font);
  letter-spacing: var(--typography-detail-m-light-letter-spacing);
  text-transform: uppercase;
  color: var(--color-token-light-content-inverse);
  width: var(--size4);
  height: var(--size4);
  line-height: var(--size4);
  border-radius: 100px;
  text-align: center;
  box-sizing: border-box;
}

.tempo-badge--small .tempo-badge__item {
  width: var(--size2);
  height: var(--size2);
  padding: var(--size1);
  font-size: 0;
}

.tempo-badge--color-dark .tempo-badge__item {
  background: var(--color-token-dark-background-inverse);
  color: var(--color-token-dark-content-inverse);
}

.tempo-badge--autolayout {
  display: flex;
  padding: 0;
  height: var(--size4);
  width: fit-content;
}

.tempo-badge--small.tempo-badge--autolayout {
  justify-content: center;
  align-items: center;
  width: var(--size4);
}

.tempo-badge--autolayout .tempo-badge__item {
  display: flex;
  padding: 0 calc(var(--size1) + 1px);
  min-width: var(--size4);
  height: var(--size4);
  width: fit-content;
}

.tempo-badge--small.tempo-badge--autolayout .tempo-badge__item {
  display: block;
  min-width: 0;
  width: var(--size2);
  height: var(--size2);
  padding: var(--size1);
  font-size: 0;
}

/* Badge colors */
.tempo-badge--color-coconut .tempo-badge__item {
  background: var(--color-coconut300);
}

.tempo-badge--color-blackberry .tempo-badge__item {
  background: var(--color-blackberry300);
}

.tempo-badge--color-cherry .tempo-badge__item {
  background: var(--color-cherry300);
}

.tempo-badge--color-apricot .tempo-badge__item {
  background: var(--color-apricot300);
}

.tempo-badge--color-kiwi .tempo-badge__item {
  background: var(--color-kiwi300);
}

.tempo-badge--color-mint .tempo-badge__item {
  background: var(--color-mint300);
}

.tempo-badge--color-water .tempo-badge__item {
  background: var(--color-water300);
}
.tempo-badge--color-blueberry .tempo-badge__item {
  background: var(--color-blueberry300);
}

.tempo-badge--color-grape .tempo-badge__item {
  background: var(--color-grape300);
}

.tempo-badge--color-watermelon .tempo-badge__item {
  background: var(--color-watermelon300);
}
