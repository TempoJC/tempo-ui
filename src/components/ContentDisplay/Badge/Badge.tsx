import cn from "classnames";
import { BadgeProps } from "./types";
import "./Badge.css";

export const BADGE_COLOR_COCONUT = "coconut";
export const BADGE_COLOR_BLACKBERRY = "blackberry";
export const BADGE_COLOR_CHERRY = "cherry";
export const BADGE_COLOR_APRICOT = "apricot";
export const BADGE_COLOR_KIWI = "kiwi";
export const BADGE_COLOR_MINT = "mint";
export const BADGE_COLOR_WATER = "water";
export const BADGE_COLOR_BLUEBERRY = "blueberry";
export const BADGE_COLOR_GRAPE = "grape";
export const BADGE_COLOR_WATERMELON = "watermelon";

/**
 * Component to render a badge (with optional text content).
 */
const Badge = ({
  className,
  color = "light",
  kind = "circular",
  size = "large",
  variant = BADGE_COLOR_CHERRY,
  testId,
  text,
}: BadgeProps) => {
  const classesContainer = cn(
    "tempo-badge",
    {
      "tempo-badge--small": size === "small",
      "tempo-badge--autolayout": kind === "autolayout",
      "tempo-badge--color-dark": color === "dark",
      "tempo-badge--color-coconut": variant === BADGE_COLOR_COCONUT,
      "tempo-badge--color-blackberry": variant === BADGE_COLOR_BLACKBERRY,
      "tempo-badge--color-cherry": variant === BADGE_COLOR_CHERRY,
      "tempo-badge--color-apricot": variant === BADGE_COLOR_APRICOT,
      "tempo-badge--color-kiwi": variant === BADGE_COLOR_KIWI,
      "tempo-badge--color-mint": variant === BADGE_COLOR_MINT,
      "tempo-badge--color-water": variant === BADGE_COLOR_WATER,
      "tempo-badge--color-blueberry": variant === BADGE_COLOR_BLUEBERRY,
      "tempo-badge--color-grape": variant === BADGE_COLOR_GRAPE,
      "tempo-badge--color-watermelon": variant === BADGE_COLOR_WATERMELON,
    },
    className,
  );

  return (
    <div className={classesContainer} data-testid={testId}>
      <div className="tempo-badge__item" data-testid={testId && `${testId}-badge`}>
        {kind === "circular" ? (text || "").slice(0, 1) : text}
      </div>
    </div>
  );
};

export default Badge;
