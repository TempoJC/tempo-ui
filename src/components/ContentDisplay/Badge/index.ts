// packages/components-badge/badge
export { default as Badge } from "./Badge";
export * from "./Badge";
