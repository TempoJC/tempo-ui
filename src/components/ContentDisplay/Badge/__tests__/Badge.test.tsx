import { render, screen } from "@testing-library/react";
import Badge from "../Badge";

describe("@tempo-ui/ContentDisplay/Badge", () => {
  it("renders", () => {
    const { asFragment } = render(<Badge text={"5"} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders small badge", () => {
    const { asFragment } = render(<Badge size="small" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders circular badge", () => {
    const { asFragment } = render(<Badge kind="circular" text="8" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("remove long text when kind is circular", () => {
    const { asFragment } = render(<Badge kind="circular" text="250" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders autolayout badge", () => {
    const { asFragment } = render(<Badge kind="autolayout" text="124" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders badge with dark color scheme", () => {
    const { asFragment } = render(<Badge text="4" color="dark" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders badge with a variant", () => {
    render(<Badge text="4" variant="coconut" testId="badge" />);
    expect(screen.getByTestId("badge")).toHaveClass("tempo-badge--color-coconut");
  });

  describe("testId prop", () => {
    it("renders the Badge with the specified data-testid attribute", () => {
      const { getByTestId } = render(<Badge testId="badge" />);
      const element = getByTestId("badge");
      expect(element).toBeInTheDocument();
      expect(element).toHaveAttribute("data-testid", "badge");
    });
  });
});
