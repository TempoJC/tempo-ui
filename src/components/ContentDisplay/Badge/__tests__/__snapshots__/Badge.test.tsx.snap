// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`@tempo-ui/ContentDisplay/Badge remove long text when kind is circular 1`] = `
<DocumentFragment>
  <div
    class="tempo-badge tempo-badge--color-cherry"
  >
    <div
      class="tempo-badge__item"
    >
      2
    </div>
  </div>
</DocumentFragment>
`;

exports[`@tempo-ui/ContentDisplay/Badge renders 1`] = `
<DocumentFragment>
  <div
    class="tempo-badge tempo-badge--color-cherry"
  >
    <div
      class="tempo-badge__item"
    >
      5
    </div>
  </div>
</DocumentFragment>
`;

exports[`@tempo-ui/ContentDisplay/Badge renders autolayout badge 1`] = `
<DocumentFragment>
  <div
    class="tempo-badge tempo-badge--autolayout tempo-badge--color-cherry"
  >
    <div
      class="tempo-badge__item"
    >
      124
    </div>
  </div>
</DocumentFragment>
`;

exports[`@tempo-ui/ContentDisplay/Badge renders badge with dark color scheme 1`] = `
<DocumentFragment>
  <div
    class="tempo-badge tempo-badge--color-dark tempo-badge--color-cherry"
  >
    <div
      class="tempo-badge__item"
    >
      4
    </div>
  </div>
</DocumentFragment>
`;

exports[`@tempo-ui/ContentDisplay/Badge renders circular badge 1`] = `
<DocumentFragment>
  <div
    class="tempo-badge tempo-badge--color-cherry"
  >
    <div
      class="tempo-badge__item"
    >
      8
    </div>
  </div>
</DocumentFragment>
`;

exports[`@tempo-ui/ContentDisplay/Badge renders small badge 1`] = `
<DocumentFragment>
  <div
    class="tempo-badge tempo-badge--small tempo-badge--color-cherry"
  >
    <div
      class="tempo-badge__item"
    />
  </div>
</DocumentFragment>
`;
